dali/kernels/slice/slice_flip_normalize_permute_pad_cpu.h

```diff
@@ -241,9 +243,79 @@ void SliceFlipNormalizePermutePadKernel(
   ));  // NOLINT
 }
 
+template <typename ExecutionEngine, int Dims, typename OutputType, typename InputType>
+void SliceFlipNormalizePermutePadKernel(
```

Comment:
Maybe you can deduplicate these functions with one slice_cpu.h, as I see the same pattern here and there.

Reply:
the implementation differs slightly so it'd be hard to generalize it and reduce duplication. It'd be rather more boilerplate. I moved some of the logic inside split_shape though.